src/components/AppModal.vue

```diff
@@ -20,14 +17,33 @@
  */
 export default {
   name: 'app-modal',
-  props: ['id'],
+  props: {
+    visible: Boolean,
+  },
+  watch: {
+    visible: {
+      handler(to) {
+        if (to) document.addEventListener('keyup', this.checkKey)
+        else document.removeEventListener('keyup', this.checkKey)
+      },
+      immediate: true,
+    },
+  },
+  destroyed() {
+    document.removeEventListener('keyup', this.checkKey)
+  },
+  methods: {
+    checkKey(e) {
+      if (e.keyCode === 27) this.$emit('close')
+    },
+  },
 }
 </script>
 
 <style lang="scss" scoped>
 .modal {
-  max-width: 80vw;
-  max-height: 84vh;
+  width: 21.875rem;
+  max-height: 37rem;
```

Comment:
This modal is a reusable component. The existing sizes are used here: https://github.com/creativecommons/cccatalog-frontend/blob/develop/src/components/MetaSearch/MetaSearchModal.vue

I would prefer if you added a `size` prop with `small/large` variants, and applied your new sizes on the 'small' version and the existing sizes on the 'large'.